pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/TabularParticipantExporter.java

```diff
@@ -0,0 +1,128 @@
+package org.broadinstitute.dsm.model.elastic.export.tabular;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import spark.Response;
+
+public abstract class TabularParticipantExporter {
+    protected static final String FILE_DATE_FORMAT = "yyyy-MM-dd";
+    public static final String TSV_FORMAT = "tsv";
+    public static final String XLSX_FORMAT = "xlsx";
+    public static final String COLUMN_REPEAT_DELIMITER = "_";
+    protected List<ModuleExportConfig> moduleConfigs;
+    protected String fileFormat;
+    protected List<Map<String, String>> participantValueMaps;
+
+    public abstract void export(Response response) throws IOException;
+
+    protected abstract String sanitizeValue(String value);
+
+    protected TabularParticipantExporter(List<ModuleExportConfig> moduleConfigs,
+                                         List<Map<String, String>> participantValueMaps, String fileFormat) {
+        this.moduleConfigs = moduleConfigs;
+        this.participantValueMaps = participantValueMaps;
+        this.fileFormat = fileFormat;
+    }
+
+    public static TabularParticipantExporter getExporter(List<ModuleExportConfig> moduleConfigs,
+                                                         List<Map<String, String>> participantValueMaps, String fileFormat) {
+        if (TSV_FORMAT.equals(fileFormat)) {
+            return new TsvParticipantExporter(moduleConfigs, participantValueMaps, fileFormat);
+        } else if (XLSX_FORMAT.equals(fileFormat)) {
+            return new ExcelParticipantExporter(moduleConfigs, participantValueMaps, fileFormat);
+        }
+        throw new RuntimeException("Unrecognized file format");
```

Comment:
I wonder if the route should have already checked that. by the time it throws here it already collected all the pt data for nothing…

Reply:
It is just factory method I believe, calculations will only start after `export` is called, so in this case I think having the check here is totally fine.

Reply:
better yet, I'll just default to tsv.  No reason to have this hard-error